src/node_file.cc

```diff
@@ -1098,7 +1098,7 @@ static void Read(const FunctionCallbackInfo<Value>& args) {
   size_t buffer_length = Buffer::Length(buffer_obj);
 
   size_t off = args[2]->Int32Value();
-  if (off >= buffer_length) {
+  if (off > buffer_length) {
```

Comment:
This should be safe because the following if-statement will catch if offset + length would be out-of-bounds of the buffer.